packages/terra-button-group/src/terra-dev-site/test/button-group/ProgrammaticDisabled.test.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import ButtonGroup from '../../../ButtonGroup';
+
+let returnButtonProps = {
+  text: 'Click to Disable',
+};
+
+const returnDisabledDefaultProps = () => {
+  returnButtonProps = {
+    isDisabled: true,
+    text: 'Disabled',
+  };
+};
+
+class ProgrammaticDisabled extends React.Component {
+  constructor() {
+    super();
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    returnDisabledDefaultProps();
+    this.forceUpdate();
```

Comment:
This is inappropriate for updating the state of the component. The function is updating an external un-scoped variable using a function callback and then forcing the component to update. 

I'd recommend moving these variables to be scoped to the instance of the component and using traditional react to update the internal state. This is a good opportunity for using hooks to simplify the example.

```jsx
import React, { useState } from 'react';
import ButtonGroup from '../../../ButtonGroup';

const ProgrammaticDisabled = () => {
  const [isDisabled, setIsDisabled] = useState(false);

  return (
    <ButtonGroup id="programmaticDisabledButtonGroup">
      <ButtonGroup.Button
        isDisabled={isDisabled}
        text={isDisabled ? 'Disabled' : 'Click to Disable'}
        id="programmaticDisabledButton"
        key="test-example-button"
        onClick={() => { setIsDisabled(!isDisabled); }}
      />
    </ButtonGroup>
  );
};

export default ProgrammaticDisabled;
```

Reply:
updated here https://github.com/cerner/terra-core/pull/2984/commits/58d1c92a99502e135e797b6507972d814e81836d.